content/source/docs/enterprise/sentinel/integrate-vcs.html.md

```diff
@@ -91,49 +91,47 @@ variable "tfe_organization" {
+Configure the `tfe` provider (version 0.3 or higher) with your API token and hostname variables.
 
 ```hcl
-variable "tfe_workspace_ids" {
-  description = "Mapping of workspace names to IDs, for easier use in policy sets."
-  type        = "map"
-
-  default = {
-    "app-prod"                = "ws-LbK9gZEL4beEw9A2"
-    "app-dev"                 = "ws-uMM93B6XrmCwh3Bj"
-  }
+provider "tfe" {
+  hostname = "${var.tfe_hostname}"
+  token    = "${var.tfe_token}"
+  version  = "~> 0.6"
 }
 ```
 
-To quickly get a list of workspace names and IDs, you can make an API call to the [List Workspaces endpoint](../api/workspaces.html#list-workspaces) and pipe the result to a `jq` command:
+### Optional: Use a Data Source to Get Workspace IDs
 
-```
-$ curl \
-  --header "Authorization: Bearer $TOKEN" \
-  --header "Content-Type: application/vnd.api+json" \
-  https://app.terraform.io/api/v2/organizations/my-organization/workspaces?page%5Bnumber%5D=1&page%5Bsize%5D=100 \
-  | jq --raw-output '.data[] | "\"\(.attributes.name)\" = \"\(.id)\""'
-```
+[inpage_ids]: #optional-use-a-data-source-to-get-workspace-ids
 
--> **Note:** If you have a very large number of workspaces, you might need to make multiple API calls to fetch subsequent pages. See [API Docs: Pagination](../api/index.html#pagination) for more details.
+The `tfe_policy_set` resource uses workspace IDs, which can be found on a workspace's [settings page](../workspaces/settings.html#id). There are several ways to work with these IDs in your configuration, all of which have advantages and disadvantages:
 
-### Configure the `tfe` Provider
+Method | Pros | Cons
+-------|------|-----
+Use literal ID strings. | Highly secure, since workspace IDs never change. | Policy set configs are opaque. Accidental misconfigurations are difficult to spot.
+Use a data source to look up IDs by name. | Convenient and easy to read. | Renaming a workspace (requires admin permissions) can remove it from a policy set.
+Use a variable to map workspace names to IDs. | Secure and easy to read. | Keeping the variable up-to-date is inconvenient.
+Use `external_id` attribute of `tfe_workspace` resources. | Secure, readable, automatically updated. | Only available if you already manage your TFE workspaces with the `tfe` Terraform provider.
 
-Configure the `tfe` provider (version 0.3 or higher) with your API token and hostname variables.
+In our examples, we want to clearly show what our policy sets are doing, which is much easier if we can refer to workspaces by name. The [`tfe_workspace_ids` data source](/docs/providers/tfe/d/workspace_ids.html) (in `tfe` provider versions ≥ 0.6) makes it easy to use workspace names:
 
 ```hcl
-provider "tfe" {
-  hostname = "${var.tfe_hostname}"
-  token    = "${var.tfe_token}"
-  version  = "~> 0.4"
+data "tfe_workspace_ids" "all" {
+  names        = ["*"]
```

Comment:
Instead of using the wildcard in the example here, I suggest to use a list of actual names that are used. Using exact names seems to indicate the expected behavior better then when using a big "catch all" hammer to just get everything by default.

Reply:
The number of names could be really large...is there some intermediate option like partial wildcard?

Reply:
No that isn't implemented at the moment.

Reply:
Haha, I remember talking about this when you were implementing the data source. 😄 

I'm using the wildcard because I'm _already_ specifying all of the workspace names I want, down in the policy set resources. Specifying them _again,_ up in the data source, seems like a waste of my time that doesn't provide me any benefit. 

Unless it somehow _does_ provide me benefit? But the only thing I can think of is if using the wildcard incurs some kind of huge multi-minute performance penalty once I hit a couple hundred workspaces. Otherwise... you say "big catch-all hammer," I hear "easy way to avoid pointless busywork." 🙂